stm32loader/main.py

```diff
@@ -266,12 +273,12 @@ def read_device_uid(self):
             return
 
         try:
-            if family != "F4":
+            try:
                 flash_size = self.stm32.get_flash_size(family)
                 device_uid = self.stm32.get_uid(family)
-            else:
-                # special fix for F4 devices
-                flash_size, device_uid = self.stm32.get_flash_size_and_uid_f4()
+            except bootloader.CommandError:
```

Comment:
CommandError can have other causes too, like flaky communication link.

I prefer to use an if-statement here.  Also to avoid a try/except inside a try/except.